src/os/bluestore/BlockDevice.h

```diff
@@ -148,6 +148,13 @@ class BlockDevice {
   virtual int invalidate_cache(uint64_t off, uint64_t len) = 0;
   virtual int open(const std::string& path) = 0;
   virtual void close() = 0;
+  virtual bool is_valid_io(uint64_t off, uint64_t len) {
```

Comment:
Right. `is_valid_io` is not getting called via an interface to be `virtual`. A simple overriding in `PMEMDevice` works well here.